### PR TITLE
Add ServiceFlags::P2P_V2

### PR DESCRIPTION
**Pull request by achow101**

### CODE REVIEW

**Review by tcharding — APPROVED** [review]
ACK 5818e04328cba73a7dc72ae47374a59dd57768e9

**Comment by tcharding** [comment]
I have no knowledge about p2p v2 but the code looks correct to me.

**Comment by tcharding** [comment]
Thanks @achow101!

**Review by sanket1729 — APPROVED** [review]
ACK 5818e04328cba73a7dc72ae47374a59dd57768e9

File: bitcoin/src/p2p/mod.rs [hunk 1]
```diff
@@ -91,6 +91,10 @@ impl ServiceFlags {
     /// See BIP159 for details on how this is implemented.
     pub const NETWORK_LIMITED: ServiceFlags = ServiceFlags(1 << 10);
 
+    /// P2P_V2 indicates that the node supports the P2P v2 encrypted transport protocol.
+    /// See BIP324 for details on how this is implemented.
+    pub const P2P_V2: ServiceFlags = ServiceFlags(1 << 11);
```
Comment (Kixunil) [inline bitcoin/src/p2p/mod.rs:96]: Checked against https://github.com/bitcoin/bips/blob/master/bip-0324.mediawiki#signaling-v2-support

**Review by Kixunil — APPROVED** [review]
ACK 5818e04328cba73a7dc72ae47374a59dd57768e9

File: bitcoin/src/p2p/mod.rs (hunk 1 above)
Comment (apoelstra) [reply bitcoin/src/p2p/mod.rs:96]: Thank you, but nobody will ever see this because I am required to click "resolve" before merging.

Comment (Kixunil) [reply bitcoin/src/p2p/mod.rs:96]: I meant it as a quick link for other reviewers to use. It also occurred to me we could click unresolve after merging but that's a bit weird and annoying.

**Pull request merged**